File: 2/2.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func isAscending(nums []int) bool {
	for i := 0; i < len(nums)-1; i++ {
		diff := nums[i+1] - nums[i]
		if diff <= 0 || diff > 3 {
			return false
		}
	}
	return true
}

func isDescending(nums []int) bool {
	for i := 0; i < len(nums)-1; i++ {
		diff := nums[i] - nums[i+1]
		if diff <= 0 || diff > 3 {
			return false
		}
	}
	return true
}

func main() {
	sum := 0
	sum2 := 0
	file, err := os.Open("input")
	if err != nil {
		log.Fatal(err)
	}
	scanner := bufio.NewScanner(file)
	defer file.Close()
	secondaryRun := [][]int{}
	for scanner.Scan() {
		strNumbers := strings.Fields(scanner.Text())
		numbers := make([]int, len(strNumbers))

		for i, str := range strNumbers {
			num, err := strconv.Atoi(str)
			if err != nil {
				log.Fatal(err)
			}
			numbers[i] = num
		}

		if isAscending(numbers) || isDescending(numbers) {
			sum++
		} else {
			secondaryRun = append(secondaryRun, numbers)
		}

		for _, nums := range secondaryRun {
			for i := range nums {
				// Create a new slice without element at index i
				withoutI := make([]int, 0, len(nums)-1)
				withoutI = append(withoutI, nums[:i]...)
				withoutI = append(withoutI, nums[i+1:]...)

				if isAscending(withoutI) || isDescending(withoutI) {
					sum2++
					break
				}
			}
		}
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	fmt.Println(sum)
	fmt.Println(sum + sum2)
}

File: 2/2.py
import os 

f = open('input', 'r').read().splitlines()

sum = 0

sum2 = 0

def ascend(a):
    return all(a[i] < a[i + 1] and 1 <= a[i+1] - a[i] and a[i+1] - a[i] <= 3  for i in range(len(a) - 1))

def descend(a):
    return all(a[i] > a[i + 1] and 1 <= a[i] - a[i+1] and a[i] - a[i+1] <= 3 for i in range(len(a) - 1))

secondrun = []
for l in f:
    a = [int(x) for x in l.split()]
    if ascend(a) or descend(a):
        sum += 1
    else:
        secondrun.append(a)

for l2 in secondrun:
    for i in range(len(l2)):
        # Create new list without element at index i
        a = l2[:i] + l2[i+1:]
        if ascend(a) or descend(a):
            sum2 += 1
            break



print(sum)
print(sum2+sum)